python/semantic_kernel/connectors/ai/open_ai/services/open_ai_text_to_image_base.py

```diff
@@ -22,16 +28,17 @@ async def generate_image(self, description: str, width: int, height: int, **kwar
         Returns:
             bytes | str: Image bytes or image URL.
         """
-        try:
-            result = await self.client.images.generate(
-                prompt=description,
-                model=self.ai_model_id,
-                size=f"{width}x{height}",  # type: ignore
-                response_format="url",
-                **kwargs,
-            )
-        except Exception as ex:
-            raise ServiceResponseException(f"Failed to generate image: {ex}") from ex
-        if not result.data or not result.data[0].url:
+        settings = OpenAITextToImageExecutionSettings(
+            prompt=description,
+            size=ImageSize(width=width, height=height),
+            ai_model_id=self.ai_model_id,
+            **kwargs,
+        )
+
+        response = await self._send_request(settings)
+
+        assert isinstance(response, ImagesResponse)  # nosec
+        if not response.data or not response.data[0].url:
```

Comment:
Do we get the "completion" tokens after image generation? We should be able to calculate the token usage based on the image size, right?

Reply:
That is unclear. The official OpenAI documents I was able to find only mentioned token usage when images were used as inputs, not when they were outputs.

And after further investigation, I think image generation is not charged by tokens. They are charged per image generation (see **Image models**): https://openai.com/api/pricing/.


Similar to audio models, users are not charged by tokens. They are charged by the duration of the audio or characters (see **Audio models**): https://openai.com/api/pricing/.